### ISSUE TITLE
linter: no-unused-vars, false positive when type param has same name as function arg

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#20670]
Given the following code:

```ts
function useParam<T>(T: number) {
    return T;
}

useParam(1);
```

Expected:
 `<T>` is unused

Actual:
no error

This happens because `T` has the same symbol id for both the type parameter AND the argument (which is a footgun imo) - typescript-eslint has this same behaviour.

**Comment by bradzacher** [comment]
This is a known issue in typescript-eslint but not something we've bothered trying to fix because it's so rare and the userland fix is a trivial "don't write dumb code - rename one of them".
